Sources/DataSources/RxHelper/ASDelegateProxy.swift

```diff
@@ -40,21 +40,27 @@ extension ObservableType {
 
                     binding(proxy, event)
 
-                    switch event {
-                    case .error(let error):
-                        bindingErrorToInterface(error)
-                        unregisterDelegate.dispose()
-                    case .completed:
-                        unregisterDelegate.dispose()
-                    default:
-                        break
+                    // ensure running unregisterDelegate.dispose() on main thread
```

Comment:
I found that following code cause #18 

```swift
let subscription = self.asObservable()
                 .observeOn(MainScheduler())
                 // and more...
```

but following does not cause same problems without your pathes.

```swift
             let subscription = self.asObservable()
                 .catchError { error in
                     bindingErrorToInterface(error)
                     return Observable.empty()
                 }
                 // source can never end, otherwise it would release the subscriber, and deallocate the data source
                 .concat(Observable.never())
                 .takeUntil(object.rx.deallocated)
                 .observeOn(MainScheduler()) // <= Move to here
```

Do you know why ? I want to make code more Rx-like if we can.